### PR TITLE
Added BitmapData and refactor

### PR DESCRIPTION
**Pull request by themuflon**
Refactored stricmp to be defined in only one place.

Refactored color float/hex conversion to a struct to reduce code duplication.

### CODE REVIEW

File: loom/graphics/gfxBitmapData.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    unsigned char* data;
+
+    // Private constructor. Use only static methods for construction.
+    BitmapData(size_t width, size_t height);
+
+public:
+
+    ~BitmapData();
+
+    // Saves the loaded data to a file. This is meant for saving diffs.
```
Comment (bengarney) [inline loom/graphics/gfxBitmapData.h]: Any restrictions on formats?

File: loom/graphics/gfxColor.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ===========================================================================
+ */
+
+#include "loom/graphics/gfxColor.h"
+
+namespace GFX
+{
+    Color::Color(float R, float G, float B, float A)
```
Comment (bengarney) [inline loom/graphics/gfxColor.cpp]: Awesome!

**Comment by bengarney** [comment]
Good solid cleanups all around.

One nit.

Please file followup card to getPixel/setPixel and upload to texture.

File: loom/graphics/gfxBitmapData.h (hunk 1 above)
Comment (themuflon) [inline loom/graphics/gfxBitmapData.h]: Currently everything is RGBA, even things supposedly reported RGB (3 channels) only... O.o

Comment (bengarney) [inline loom/graphics/gfxBitmapData.h]: Hmm, that's worth a followup. But I mean - can we save as JPEG, PNG,
RAW...? What library determines it?

Comment (themuflon) [inline loom/graphics/gfxBitmapData.h]: We can save as BMP, PNG and TGA. This is determined by stbi - it supports more filetypes for loading than for saving.

I think HDR is in there for saving too, but is skipped that.

Comment (bengarney) [inline loom/graphics/gfxBitmapData.h]: OK. Please document that in a comment so people know.

**Comment by bengarney** [comment]
Awesome.

**Pull request merged**